pmd-core/src/main/java/net/sourceforge/pmd/util/FileUtil.java

```diff
@@ -148,21 +154,27 @@ public static boolean findPatternInFile(final File file, final String pattern) {
         Pattern regexp = Pattern.compile(pattern);
         Matcher matcher = regexp.matcher("");
 
-        FileIterable it = new FileIterable(file);
-        for (String line : it) {
-            matcher.reset(line); // reset the input
-            if (matcher.find()) {
-                return true;
+        try {
+            for (String line : Files.readAllLines(file.toPath())) {
```

Comment:
this call requires a charset in Java 7. The single arg version was added in Java 8 and is unavailable in master